docs/src/main/paradox/scala/http/routing-dsl/directives/custom-directives.md

```diff
@@ -169,8 +169,17 @@ A `Directive[(String)]` extracts one String value (like the hostName directive).
 type Directive1[T] = Directive[Tuple1[T]]
 ```
 
-A `Directive[(Int, String)]` extracts an `Int` value and a `String` value
-(like a `parameters('a.as[Int], 'b.as[String])` directive).
+A `Directive[(String, Int)]` extracts a `String` value and an `Int` value
+(like a `parameters('a.as[String], 'b.as[Int])` directive). Such a directive can be defined to extract the 
+hostname and port of a request:
+
+@@snip [CustomDirectivesExamplesSpec.scala]($test$/scala/docs/http/scaladsl/server/directives/CustomDirectivesExamplesSpec.scala) { #scratch-1 }
+
+Beside using `Directive.apply`, you can also extending `Directive` directly:
+
+@@snip [CustomDirectivesExamplesSpec.scala]($test$/scala/docs/http/scaladsl/server/directives/CustomDirectivesExamplesSpec.scala) { #scratch-2 }
+
+However, extending `Directive` directly is actually uncommon and the first is preferable for common use cases.
```

Comment:
Perhaps move this caveat above the example? It seems a bit 'hidden' now

Reply:
Good advice, an early warning will be helpful.